### PR TITLE
use primitive root instead of prime root

### PR DESCRIPTION
**Pull request by slaperche-scality**
This will avoid confusion.

Closes: #169

### CODE REVIEW

File: src/gf_ring.h [hunk 1]
```diff
@@ -490,30 +490,55 @@ T RingModN<T>::weak_rand(void)
...
- *  Methodology: checking its order is not a proper divisor of (q-1)
- * 1. Prime factorization (q - 1) = p1^r1 * p2^r2 * ... pm^rm
- * 2. Get "necessary" proper divisors of (q - 1):
- *      D = {(q-1)/p1, (q-1)/p2, .., (q-1)/pm }
- * 3. x is a primitive root if for every divisor d of D:
- *                          x^d != 1
+/** Check if a number is primitive root or not.
+ *
+ * A number `x` is a primitive root if its order \f$y = q - 1\f$, i.e.
+ * \f$x^{i}\neq 1\qquad (i=1, 2, 3, \dots , q-1)\f$
```
Comment (lamphamsy) [inline src/gf_ring.h]: Should it be `.., \dots , q-2)\f$`?

File: src/gf_ring.h [hunk 2]
```diff
@@ -490,30 +490,55 @@ T RingModN<T>::weak_rand(void)
...
- *      x^((q-1)/y) = 1 => (x^((q-1)/y))^(y/p1) = 1 <=> x^((q-1)/p1) = 1.
- *  Contradict to Step 3 of the algorith.
+ *
+ *  Suppose that `x` satisfies the algorithm but its order is a proper divisor
+ *  `y` of `q-1`.
+ *
+ *  This order can be expressed as \f$\frac{q-1}{y}\f$ where:
+ *
+ *  \f[
+ *    y = p_1^{r_1} p_2^{r_2} \dots p_m^{r_m}
```
Comment (lamphamsy) [inline src/gf_ring.h]: It should be `y = p_1^{s_1} p_2^{s_2} \dots p_m^{s_m}`
with an implicit note: `s_i <= r_i`

File: src/gf_ring.h [hunk 3]
```diff
@@ -490,30 +490,55 @@ T RingModN<T>::weak_rand(void)
...
+ *  \f]
+ *
+ *  and
+ *
+ *  \f[
+ *    x^{\frac{q-1}{y}} = 1
+ *  \f]
+ *
+ *  Withouth loss of generality, we suppose s1 ≥ 1. We have
```
Comment (lamphamsy) [inline src/gf_ring.h]: It should be `s_1 ≥ 1`

File: src/gf_ring.h [hunk 4]
```diff
@@ -534,24 +559,29 @@ bool RingModN<T>::is_prime_root(T nb)
     return ok;
 }
 
-/*
- * Find primitive root of finite field. A number x is a primitive root if its
- *  order y = q - 1, i.e. x^i != 1 for every i < q-1
+/** Find primitive root of finite field.
+ *
+ * A number `x` is a primitive root if its order \f$y = q - 1\f$, i.e.
+ * \f$x^{i}\neq 1\qquad (i=1, 2, 3, \dots , q-1)\f$
```
Comment (lamphamsy) [inline src/gf_ring.h]: Same above, should it be .., \dots , q-2)\f$?

File: src/gf_ring.h (hunk 2 above)
Comment (slaperche-scality) [reply src/gf_ring.h]: Indeed, bad copy paste, I'll fix it!

File: src/gf_ring.h (hunk 1 above)
Comment (slaperche-scality) [reply src/gf_ring.h]: Good catch, I'll fix it

File: src/gf_ring.h (hunk 4 above)
Comment (slaperche-scality) [reply src/gf_ring.h]: Done

File: src/gf_ring.h (hunk 3 above)
Comment (slaperche-scality) [reply src/gf_ring.h]: Done

**Pull request merged**